tests/catalog/test_rest_serializers.py

```diff
@@ -0,0 +1,159 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+from pyiceberg.catalog.rest.expression import (
+    AndOrExpression,
+    Expression,
+    LiteralExpression,
+    Term,
+)
+from pyiceberg.catalog.rest.planning_models import (
+    DataFile,
+    DeleteFile,
+    EqualityDeleteFile,
+    FileScanTask,
+    PlanTableScanRequest,
+    PositionDeleteFile,
+    ScanTasks,
+)
+
+
+def test_serialize_plan_table_scan_request() -> None:
+    """Test serializing a PlanTableScanRequest to a dict"""
+    expression = AndOrExpression(
+        type="and",
+        left=Expression(
+            root=AndOrExpression(
+                type="or",
+                left=Expression(
+                    root=AndOrExpression(
+                        type="and",
+                        left=Expression(
+                            root=LiteralExpression(type="lt", term=Term(root="a"), value={"type": "integer", "value": 1})
+                        ),
+                        right=Expression(
+                            root=LiteralExpression(type="lt-eq", term=Term(root="b"), value={"type": "integer", "value": 2})
+                        ),
+                    )
+                ),
+                right=Expression(root=LiteralExpression(type="eq", term=Term(root="c"), value={"type": "integer", "value": 3})),
+            )
+        ),
+        right=Expression(root=LiteralExpression(type="gt", term=Term(root="d"), value={"type": "integer", "value": 4})),
+    )
+    request = PlanTableScanRequest(
+        snapshot_id=1,
+        select=["a", "b", "c"],
+        filter=Expression(root=expression),
+        case_sensitive=True,
+    )
+    assert request.model_dump_json(exclude_none=True) == snapshot_json_for_plan_table_scan_request()
+
+
+def test_deserialize_plan_table_scan_request() -> None:
+    """Test deserializing a PlanTableScanRequest from a dict"""
```

Comment:
Where is the dict?

Reply:
It's down below. I've changed the tests + added comments to make it more obvious.